### PR TITLE
BL-683/Random-Generator-Validation

### PR DESCRIPTION
**Pull request by jack-semder**
## Description

We've added a testing file that checks our random generators against our validation schema. This test is called in the seeds file, so that if one test fails, the entire seeding process stops. 

We've used simple assert statements to test these classes. 

- [x] New feature

## Checklist:

- [x] My code follows PEP8 style guide
- [x] I have removed unnecessary print statements from my code
- [x] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

[LOOM VID x1000](https://www.loom.com/share/6bd8f27dd8114416bad65b4f4d52a77c)

### CODE REVIEW

File: test_coverage/schema_validation_testing.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    meeting = RandomMeeting(generate_uuid(16), generate_uuid(16))
+    meeting_dict = vars(meeting)
+
+    assert Meeting(**meeting_dict)
+
+
+def test_feedback():
+    feedback = RandomMenteeFeedback(generate_uuid(16), generate_uuid(16))
+    feedback_dict = vars(feedback)
+    Feedback(**feedback_dict)
```
Comment (BrokenShell) [inline test_coverage/schema_validation_testing.py:31]: Line 31 is redundant, the assertion takes care of it.

**Review by BrokenShell — CHANGES_REQUESTED** [review]
This is a great start! However, I do see a tiny bit of room for improvement. Instead of just testing each random generator once, lets test them each 1000 times. Something like...
```assert all(Mentor(**vars(RandomMentor()) for _ in range(1000))```

**Review by L0lbrich — APPROVED** [review]
Great start, testing is a very important part of this process, especially when randomly creating new entries.

**Review by L0lbrich — APPROVED** [review]
Looks good after changes

File: test_coverage/schema_validation_testing.py [hunk 2]
```diff
@@ -3,34 +3,26 @@
...
-    assert Mentor(**mentor_dict)
+    assert all(Mentor(**vars(RandomMentor()) for _ in range(1000)))
 
 
 def test_mentee():
     mentee = RandomMentee()
-    mentee_dict = vars(mentee)
 
-    assert Mentee(**mentee_dict)
+    assert all(Mentee(**vars(mentee) for _ in range(1000)))
```
Comment (BrokenShell) [inline test_coverage/schema_validation_testing.py:13]: This tests the same RandomMentee instance 1000 times. The RandomMentor  above is correct. Check meeting and feedback.

**Comment by BrokenShell** [comment]
> Looks good after changes

Almost!

File: test_coverage/schema_validation_testing.py (hunk 2 above)
Comment (woodbenwood) [reply test_coverage/schema_validation_testing.py]: Just seeing this, looking into it presently

File: data_generators/generators.py [hunk 3]
```diff
@@ -38,6 +38,11 @@ def __init__(self):
         self.is_active = percent_true(80)
         self.accepting_new_mentees = percent_true(33)
 
+#
```
Comment (BrokenShell) [inline data_generators/generators.py:41]: let's remove the commented out code

**Comment by BrokenShell** [comment]
This is looking really good. One thing left to do: remove the commented out code in generators.py

**Comment by woodbenwood** [comment]
All set.

**Comment by woodbenwood** [comment]
added [Loom vid](https://www.loom.com/share/6bd8f27dd8114416bad65b4f4d52a77c)eo to og description

**Review by BrokenShell — APPROVED** [review]

**Comment by BrokenShell** [comment]
This is fantastico

**Review by CrisIV — APPROVED** [review]
Testing is important and it looks like all the changes requested have been made, looks great!

**Review by L0lbrich — APPROVED** [review]
Looks much better after changes!

**Pull request merged**